packages/main/src/RPA/JavaAccessBridge.py

```diff
@@ -0,0 +1,520 @@
+        Write text into Swing application
+            Start Process    java -jar BasicSwing.jar    shell=${TRUE}    cwd=${CURDIR}
+            Select Window    Chat Frame
+            Type Text    role:text    text for the textarea
+            Type Text    role:text    text for the input field  index=1  clear=${TRUE}
+            Click Element    role:push button and name:Send
+
+    .. code:: python
+
+        from RPA.JavaAccessBridge import JavaAccessBridge
+        import subprocess
+
+        jab = JavaAccessBridge()
+
+        subprocess.Popen(["java", "-jar", "BasicSwing.jar"], shell=True, cwd=".", close_fds=True)
+        jab.select_window("Chat Frame")
+        jab.type_text("role:text", "text for the textarea", enter=True)
+        jab.type_text("role:text", "text for the input field", index=1, clear=True)
+        jab.click_element("role:push button and name:Send")
+    """  # noqa: E501, W605
+
+    def __init__(self):
+        self.logger = logging.getLogger(__name__)
+        self.version_printed = False
+        self.jab_wrapper = None
+        self.context_info_tree = None
+        self.pumper_thread = None
+        self._initialize()
+        self.refresh_counter = 1
+        self.display_scale_factor = (
+            ctypes.windll.shcore.GetScaleFactorForDevice(0) / 100
+        )
+
+    def _initialize(self):
+        pipe = queue.Queue()
+        self.pumper_thread = threading.Thread(
+            target=self._pump_background, daemon=True, args=[pipe]
+        )
+        self.pumper_thread.start()
+        self.jab_wrapper = pipe.get()
+        if not self.jab_wrapper:
+            raise Exception("Failed to initialize Java Access Bridge Wrapper")
+        time.sleep(1)
+        atexit.register(self._handle_shutdown)
+
+    def _handle_shutdown(self):
+        if self.jab_wrapper:
+            self.jab_wrapper.shutdown()
+
+    def _pump_background(self, pipe: queue.Queue):
+        try:
+            jab_wrapper = JavaAccessBridgeWrapper()
+            pipe.put(jab_wrapper)
+            message = byref(wintypes.MSG())
+            while GetMessage(message, 0, 0, 0) > 0:
+                TranslateMessage(message)
+                self.logger.debug("Dispatching msg=%s", repr(message))
+                DispatchMessage(message)
+        # pylint: disable=broad-except
+        except Exception as err:
+            self.logger.error(err)
+            pipe.put(None)
+        finally:
+            self.logger.info("Stopped processing events")
+
+    @keyword
+    def select_window(
+        self, title: str, bring_foreground: bool = True, timeout: int = 30
+    ):
+        """Selects Java application window as target for the automation
+
+        :param title: application window title
+        :param bring_foreground: if application is brought to foreground or not
+        :param timeout: selection timeout
+        """
+        window_found = False
+        interval = float(0.5)
+        end_time = time.time() + float(timeout)
```

Comment:
might be a case of polishing a turd.... even all internal RF timeouts use non-monotonic time.

Reply:
Still no reason for us not to use monotonic time. They can fix the RF side at some point and then we would have to fix our libraries. Not saying we should do this for all libraries, but at least the new ones we create? Libraries also can be used as pure python code.

Reply:
sure, just saying. using `time.monotonic` is probably better if it doesn't need to be compatible with `time.time`. we have 15ish places in rpaframework where this would have to be fixed as well.

Reply:
I can't see an official spec that `time.sleep` uses monotonic time, but the cpython implementation happens to be

Reply:
maybe something that could be changed cross rpaframework at a later date ?

Reply:
Yeah, maybe over engineering on my part, just gotten used to use the monotonic time API with different languages.